creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -274,15 +291,112 @@ class SnapbackSM {
+      const newSecondary = await this.determineNewReplicaSet({ wallet, secondary1, secondary2, primary })
+
+      this.log(`[issueUpdateReplicaSetOp] Updating userId=${userId} wallet=${wallet} replica set=[${primary},${secondary1},${secondary2}] to new replica set=[${primary},${currentHealthySecondary},${newSecondary}]`)
+
+      if (!this.snapbackReconfigEnabled) return
+
+      // Create new array of replica set spIds and write to URSM
+      phase = issueUpdateReplicaSetOpPhases.UPDATE_URSM_REPLICA_SET
+      newReplicaSetSPIds = [
+        this.endpointToSPIdMap[primary],
+        this.endpointToSPIdMap[currentHealthySecondary],
+        this.endpointToSPIdMap[newSecondary]
+      ]
+
+      await this.audiusLibs.contracts.UserReplicaSetManagerClient.updateReplicaSet(
+        userId,
+        newReplicaSetSPIds[0], // primary
+        newReplicaSetSPIds.slice(1) // [secondary1, secondary2]
+      )
+
+      // Sync data from existing primary to new secondary
+      phase = issueUpdateReplicaSetOpPhases.ENQUEUE_SYNCS
+      await this.enqueueSync({
+        ...baseSyncRequestParams,
+        primaryEndpoint: primary,
+        secondaryEndpoint: newSecondary
+      })
+    } catch (e) {
+      const errorMsg = `[issueUpdateReplicaSetOp] userId=${userId} wallet=${wallet} failed at phase=${phase} reconfiguring to spIds=[${newReplicaSetSPIds}]: ${e.toString()}\n${e.stack}`
+      this.logError(errorMsg)
+      throw new Error(errorMsg)
+    }
+  }
+
+  /**
+   * @param {Object} param
+   * @param {string[]} param.unhealthyReplicasSet an array of endpoints of unhealthy replica set nodes
+   * @param {string} param.primary current user's primary endpoint
+   * @param {string} param.secondary1 current user's first secondary endpoint
+   * @param {string} param.secondary2 current user's second secondary endpoint
+   * @returns {string} the endpoint of the newly selected secondary. Should not be a node in the
+   *        current replica set.
    */
-  async issueUpdateReplicaSetOp (userId, wallet, primary, secondary1, secondary2, unhealthyReplica) {
-    // await libs.userReplicaSetManagerClient._updateReplicaSet(userId, primary, [new secondaries], primary, [old secondaries])
+  async determineNewReplicaSet ({ primary, secondary1, secondary2, wallet }) {
+    // Fetch all the healthy nodes while disabling sync checks to select nodes for new replica set
+    if (!this.healthyNodes) {
+      const { services: healthyServicesMap } = await this.audiusLibs.ServiceProvider.autoSelectCreatorNodes({
+        performSyncCheck: false
+      })
+
+      this.healthyNodes = Object.keys(healthyServicesMap)
+      if (this.healthyNodes.length === 0) throw new Error('Unable to query any healthy Content Nodes')
+    }
+
+    // Make sure current replica set will not be chosen as potential nodes for the new replica set
+    const healthyPotentialPeers = this.healthyNodes.filter(node => node !== primary && node !== secondary1 && node !== secondary2)
```

Comment:
nit but cleaner to do `this.healthyNodes.filter(node => ![primary, secondary1, secondary2].includes(node))`